### PR TITLE
Seaparate CoinSwitch as a plugin

### PR DESCRIPTION
**Pull request by Kukks**
This partially decouples coinswitch  from BTCPay. There are some some dependencies which require the code to remain inside the main codebase, but we can abstract that in a later PR. This PR includes #2384 as it makes Coinswitch appear as an option in the Integrations page.

### CODE REVIEW

**Comment by dstrukt** [comment]
😍

**Comment by Eskyee** [comment]
💯

File: BTCPayServer.Tests/CoinSwitchTests.cs [hunk 1]
```diff
@@ -1,91 +1,91 @@
...
-
-                store = await tester.PayTester.StoreRepository.FindStore(user.StoreId);
-
-                Assert.False(store.GetStoreBlob().CoinSwitchSettings.Enabled);
-            }
-        }
-
-    }
-}
+// using System.Threading.Tasks;
```
Comment (NicolasDorier) [inline BTCPayServer.Tests/CoinSwitchTests.cs:1]: remove those?

**Comment by pavlenex** [comment]
@Kukks What's the goal with this one?
Extract it as a system plugin, meaning I can't disable it at all?
I don't think this should be a system plugin, but I assume you're doing this to not break stuff, right?

Screenshot 2021-03-31 at 19 08 31

**Comment by Kukks** [comment]
> ~@Kukks What's the goal with this one?~
> ~Extract it as a system plugin, meaning I can't disable it at all? I don't think this should be a system plugin, but I assume you're doing this to not break stuff, right?~
> 
> NVM. I now understand it. It's a cool concept, would be great to further abstract this in the future.
> 
> Screenshot 2021-03-31 at 19 08 31
> 
> Screenshot 2021-03-31 at 19 11 34
> 
> Concept ACK.

Yes, it's a system plugin until I can develop the plugin system further enough to completely make it opt in. Also need to think about how to migrate users already using this feature.

File: BTCPayServer.Tests/CoinSwitchTests.cs (hunk 1 above)
Comment (rockstardev) [reply BTCPayServer.Tests/CoinSwitchTests.cs]: ae93cfa

**Comment by NicolasDorier** [comment]
Nice

**Pull request merged**